eng/pipelines/runtime.yml

```diff
@@ -75,132 +102,354 @@ jobs:
     - OSX_x64
     - Linux_x64
     - Linux_arm
+    - Linux_arm64
+    - Linux_musl_x64
     - Windows_NT_x86
     - Windows_NT_x64
     - Windows_NT_arm
     - Windows_NT_arm64
     jobParameters:
       testGroup: innerloop
+      condition: or(eq(dependencies.checkout.outputs['SetPathVars_coreclr.containsChange'], true), eq(variables['isFullMatrix'], true))
+
 
 #
-# Build libraries using live CoreLib from CoreCLR
+# Build CoreCLR release
+# Only when CoreCLR is changed
+# Always as they are needed by Installer and we always build and test the Installer.
 #
 - template: /eng/pipelines/common/platform-matrix.yml
   parameters:
-    jobTemplate: /eng/pipelines/libraries/build-job.yml
-    buildConfig: Release
+    jobTemplate: /eng/pipelines/coreclr/templates/build-job.yml
+    buildConfig: release
     platforms:
     - OSX_x64
     - Linux_x64
     - Linux_arm
+    - Linux_arm64
+    - Linux_musl_x64
+    - Linux_musl_arm64
+    - Windows_NT_x64
     - Windows_NT_x86
+    - Windows_NT_arm
+    - Windows_NT_arm64
+    jobParameters:
+      testGroup: innerloop
+
+#
+# Build CoreCLR Formatting Job
+# Only when CoreCLR is changed
+#
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/coreclr/templates/format-job.yml
+    platforms:
+    - Linux_x64
     - Windows_NT_x64
+    jobParameters:
+      condition: or(eq(dependencies.checkout.outputs['SetPathVars_coreclr.containsChange'], true), eq(variables['isFullMatrix'], true))
+
+#
+# Build libraries using live CoreLib
+# These set of libraries are built always no matter what changed
+# The reason for that is because Corelib and Installer needs it and
+# These are part of the test matrix for Libraries changes.
+#
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/libraries/build-job.yml
+    buildConfig: Release
+    platforms:
+    - Linux_arm
+    - Linux_musl_arm64
     - Windows_NT_arm
     - Windows_NT_arm64
+    - Windows_NT_x86
+    jobParameters:
+      liveCoreClrBuildConfig: release
+
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/libraries/build-job.yml
+    buildConfig: ${{ variables.debugOnPrReleaseOnRolling }}
+    platforms:
+    - Linux_arm64
+    - Linux_musl_x64
+    - Linux_x64
+    - OSX_x64
+    - Windows_NT_x64
+    jobParameters:
+      liveCoreClrBuildConfig: release
+#
+# Libraries Build that only run when libraries is changed
```

Comment:
Shouldn't this be "libraries tests" and not "build"? The expectation is that  libraries build on every change but only tests run based on the content of the change

Reply:
No, this is WebAssembly, allConfigurations and NETFX, which doesn't make sense to always build as they don't provide value other than when libraries is changed.

When mono is consolidated and we start running tests for WebAssembly, then it will make sense to build WebAssembly always. 

Said that, WebAssembly is not an expensive operation as it only builds and takes around 10-15 mins, but I was trying to be aggressive on minimizing the number of agents we use for each build.